Components/Queues/Drivers/PhpAmqpLibDriver.php

```diff
@@ -0,0 +1,333 @@
+use Smartbox\Integration\FrameworkBundle\Core\Messages\Context;
+use Smartbox\Integration\FrameworkBundle\DependencyInjection\Traits\UsesSerializer;
+use Smartbox\Integration\FrameworkBundle\Service;
+
+/**
+ * Class PhpAmqpLibDriver.
+ */
+class PhpAmqpLibDriver extends Service implements AsyncQueueDriverInterface
+{
+    use UsesSerializer;
+
+    /**
+     * This field specifies the prefetch window size in octets.
+     * The server will send a message in advance if it is equal to or smaller in size than the available prefetch size
+     * (and also falls into other prefetch limits).
+     * May be set to zero, meaning "no specific limit", although other prefetch limits may still apply.
+     * The prefetch-size is ignored if the no-ack option is set.
+     */
+    const PREFETCH_SIZE = null;
+
+    /**
+     * Represents the amount of message to consume by iteration
+     */
+    const PREFETCH_COUNT = 10;
+
+    /*
+     * Default exchange name
+     */
+    const EXCHANGE_NAME = '';
+
+    /**
+     * Default port to connection
+     */
+    const DEFAULT_PORT = 5672;
+
+    /**
+     * @var string
+     */
+    protected $format = AsyncQueueDriverInterface::FORMAT_JSON;
+
+    /**
+     * @var AMQPChannel|null
+     */
+    private $channel;
+
+    /**
+     * @var \AMQPExchange|null
+     */
+    private $exchange;
+
+    /**
+     * @var array|null
+     */
+    protected $connectionsData;
+
+    /**
+     * @var AMQPStreamConnection
+     */
+    protected $stream;
+
+    /**
+     * Number of the port used by the broker
+     *
+     * @var int $port
+     */
+    protected $port;
+
+    public function __destruct()
+    {
+        $this->destroy(null);
+    }
+
+    /**
+     * Method responsible to catch the parameters to configure the driver.
+     *
+     * {@inheritDoc}
+     */
+    public function configure(string $host, string $username, string $password, string $vhost = null)
+    {
+        $this->connectionsData[] = [
+            'host' => $host,
+            'user' => $username,
+            'password' => $password,
+            'vhost' => $vhost,
+            'port' => $this->port
+        ];
+    }
+
+    /**
+     * Method responsible to open the connection with the broker.
+     *
+     * @param bool $shuffle - used to randomize the connections if exists more than one
+     *
+     * @throws \Exception
+     */
+    public function connect($shuffle = true)
+    {
+        $this->shuffleConnections($shuffle);
+        try {
+            $this->stream = AMQPStreamConnection::create_connection($this->connectionsData, []);
+        } catch (AMQPIOException $exception) {
+            throw new \AMQPConnectionException($exception->getMessage());
+        }
+    }
+
+    /**
+     * Responsible to disconnect with the broker.
+     *
+     * @throws \Exception
+     * @return void
+     */
+    public function disconnect()
+    {
+        if ($this->stream instanceof AbstractConnection) {
```

Comment:
Here you could use the method `validateConnection()` you have created in this class

Reply:
Applied.